src/pudl_archiver/archivers/ferc/ferccid.py

```diff
@@ -0,0 +1,88 @@
+"""Archive FERC Central Identifier (CID) data."""
+
+import re
+from pathlib import Path
+from urllib.parse import unquote, urljoin, urlparse
+
+from playwright.async_api import async_playwright
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+
+SOURCE_CSV_URL = "https://ferc.gov/media/ferc-cid-listing-csv"
+SOURCE_XLSX_URL = "https://www.ferc.gov/media/ferc-cid-listing"
+
+
+BASE_URL = "https://ferc.gov/sites/default/files/"
+
+
+class FercCIDArchiver(AbstractDatasetArchiver):
+    """FERC CID archiver."""
+
+    name = "ferccid"
+
+    async def get_resources(self) -> ArchiveAwaitable:
+        """Download FERC CID resources."""
+        # get CSV
+        # https://ferc.gov/sites/default/files/2025-06/FERC CID Listing 6-10-2025.csv
```

Comment:
I'm suspicious that this filename (and the XLSX one below) may be manually assigned, and not follow a consistent naming convention between releases, in which case we may want to use a more flexible regular expression.

This would also make the extraction of the date elements more complex / unreliable. It might be possible to use `pd.to_datetime()` on a subset of the string and have it to more generic parsing.

Reply:
I think their interactive table is a better primary source for this data because:

- It's been updated much more recently, and I suspect more consistently.
- It seems like the CSV/XLSX outputs are the same (while the lonely individual files are not)
- The lonely CSV/XLSX filenames look like they are probably not stable.
- The interactive interface also provides a data dictionary, which we should archive.
- The interactive interface is uniform across many interesting tables that it would be great to grab.

I haven't really worked with Playwright, but there are other archivers in here that use it to navigate javascripty interfaces.